clarity_epp/export/illumina.py

```diff
@@ -1,53 +1,111 @@
 """Illumina export functions."""
 import operator
 import re
-import csv
 
-from genologics.entities import Process, Artifact
+from genologics.entities import Process
 
 from .. import get_sequence_name, get_sample_artifacts_from_pool
-import clarity_epp.export.utils
+from clarity_epp.export.utils import get_sample_sequence_index, reverse_complement
 import config
 
 
-def update_samplesheet(lims, process_id, artifact_id, output_file, conversion_tool):
-    """Update illumina samplesheet."""
-    process = Process(lims, id=process_id)
-    trim_last_base = True  # Used to set Read1EndWithCycle
+def get_project(projects, urgent=False):
+    """Get a project name for sample."""
+    if urgent:  # Sort projects for urgent samples on name
+        projects_sorted = sorted(projects.items(), key=operator.itemgetter(0))
+        for project in projects_sorted:
+            if project[1] < 9:
+                return project[0]  # return first project with < 9 samples
+
+    # Sort projects on number of samples, if not urgent or no projects left with <9 samples
+    projects_sorted = sorted(projects.items(), key=operator.itemgetter(1))
+    return projects_sorted[0][0]  # return project with least amount of samples.
+
+
+def get_override_cycles(read_len, umi_len, index_len, max_index_len, index_2_orientation):
+    """Get override cycles per sample."""
+    read_cycles = ['', '']
+    index_cycles = ['', '']
+
+    for idx in range(len(read_cycles)):
```

Comment:
> Is it possible to have different numbers of index_cycles to the read_cycles?

Yes but you always need to set them both currently. So yes in theory we can merge the two for loops into one loop, but I am not sure whether I would like to do so. It might confuse readers of the code because we use (for example) `len(read_cyles)` to also set `index_cycles`. If we ever in or decrease on of the two you also need to make sure to fix the loop again...

In terms of performance it does not really matter...